qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,295 @@
+                                           params.get("netperf_client_link"))
+        client_path = params.get("client_path")
+
+        n_client = utils_netperf.NetperfClient(
+            vm.get_address(), client_path,
+            netperf_source=netperf_client_link,
+            client=params.get("shell_client"),
+            port=params.get("shell_port"), username=params.get("username"),
+            password=params.get("password"), prompt=params.get("shell_prompt"),
+            linesep=params.get("shell_linesep", "\n").encode().decode(
+                'unicode_escape'),
+            status_test_command=params.get("status_test_command", ""),
+            compile_option=params.get("compile_option", ""))
+
+        n_server = utils_netperf.NetperfServer(
+            utils_net.get_host_ip_address(params),
+            params.get("server_path", "/var/tmp"),
+            netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                        params.get("netperf_server_link")),
+            password=params.get("hostpassword"),
+            compile_option=params.get("compile_option", ""))
+
+        try:
+            n_server.start()
+            # Run netperf with message size defined in range.
+            netperf_test_duration = int(
+                params.get("netperf_test_duration", 180))
+            netperf_para_sess = params.get("netperf_para_sessions", "1")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_cmd_prefix = params.get("netperf_cmd_prefix", "")
+            netperf_output_unit = params.get("netperf_output_unit")
+            netperf_package_sizes = params.get("netperf_sizes")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if params.get("netperf_remote_cpu") == "yes":
+                test_option += " -C"
+            if params.get("netperf_local_cpu") == "yes":
+                test_option += " -c"
+            if netperf_output_unit in "GMKgmk":
+                test_option += " -f %s" % netperf_output_unit
+            start_time = time.time()
+            stop_time = start_time + netperf_test_duration
+            num = 0
+            for protocol in test_protocols.split():
+                t_option = "%s -t %s" % (test_option, protocol)
+                n_client.bg_start(utils_net.get_host_ip_address(params),
+                                  t_option,
+                                  netperf_para_sess, netperf_cmd_prefix,
+                                  package_sizes=netperf_package_sizes)
+                if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                       "Wait netperf test start"):
+                    logging.info("Netperf test start successfully.")
+                else:
+                    test.error("Can not start netperf client.")
+                num += 1
```

Comment:
Meaningless iteration.

Reply:
I have solved the above problem, please help me review it again,thanks a lot.